Divine/CLI/CommandLineArguments.cs

```
@@ -0,0 +1,327 @@
+            DefaultValue = null,
+            AllowedValues = "dae;gr2;lsv;pak;lsj;lsx;lsb;lsf",
+            ValueOptional = false,
+            Optional = true
+        )]
+        public string OutputFormat;
+
+        // @formatter:off
+        [EnumeratedValueArgument(typeof(string), 'p', "package-version",
+            Description = "Set package version",
+            DefaultValue = "v13",
+            AllowedValues = "v7;v9;v10;v13",
+            ValueOptional = false,
+            Optional = true
+        )]
+        public string PackageVersion;
+
+        // @formatter:off
+        [ValueArgument(typeof(string), 's', "source",
+            Description = "Set source file path or directory",
+            DefaultValue = null,
+            ValueOptional = false,
+            Optional = false
+        )]
+        public string Source;
+
+        // @formatter:on
+
+        public static LogLevel GetLogLevelByString(string optionLogLevel)
+        {
+            LogLevel logLevel;
+
+            switch (optionLogLevel)
+            {
+                case "off":
+                    logLevel = Enums.LogLevel.OFF;
+                    break;
+                case "fatal":
+                    logLevel = Enums.LogLevel.FATAL;
+                    break;
+                case "error":
+                    logLevel = Enums.LogLevel.ERROR;
+                    break;
+                case "warn":
+                    logLevel = Enums.LogLevel.WARN;
+                    break;
+                case "info":
+                    logLevel = Enums.LogLevel.INFO;
+                    break;
+                case "debug":
+                    logLevel = Enums.LogLevel.DEBUG;
+                    break;
+                case "trace":
+                    logLevel = Enums.LogLevel.TRACE;
+                    break;
+                case "all":
+                    logLevel = Enums.LogLevel.ALL;
+                    break;
+                default:

```

In the Get*ByString functions we should probably throw an exception if an incorrect value is specified instead of returning an arbitrary default. (Or at least return the default only if the input is "empty").

---

You can't pass an incorrect value. Argument values are constrained.